jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/model/solver/TestSolver.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.task.assigning.model.solver;
+
+import java.util.List;
+
+import org.jbpm.task.assigning.BaseTaskAssigningTest;
+import org.jbpm.task.assigning.model.Task;
+import org.jbpm.task.assigning.model.TaskAssigningSolution;
+import org.jbpm.task.assigning.model.User;
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+
+import static org.jbpm.task.assigning.model.solver.TaskHelper.extractTaskList;
+import static org.jbpm.task.assigning.model.solver.TaskHelper.isPotentialOwner;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class TestSolver extends BaseTaskAssigningTest {
+
+    private static final long MILLISECONDS_TIME_SPENT_LIMIT = 20000;
```

Comment:
correct, it's a "good enough" value to get a feasible solution. (On the basis that It's know by construction of the model and the input data set that a feasible solution can be produced).

Reply:
The reason I asked is that test using this type of termination might be unstable: once it runs in slower environment, 20 seconds might not be sufficient (and also the other way - on a faster computer it may do unnecessary computation).

If we use OptaPlanner in a reproducible mode, each time it will take the same number of steps to reach the same solution. Thus I would rather use step count as a termination condition.